roles/ceph-dashboard/tasks/set_api_url_host.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: set grafana url
+  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-url {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ grafana_port }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  run_once: true
+  changed_when: false
+
+- name: set alertmanager host
+  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-alertmanager-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ alertmanager_port }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  run_once: true
+  changed_when: false
+
+- name: set prometheus host
+  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-prometheus-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ prometheus_port }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  run_once: true
+  changed_when: false
```

Comment:
You only needed to iterate through grafana servers (as the `inject grafana dashboard layouts` tasks is applied *only* to the Grafana instance pointed by `grafana-api-url`). Doing the same for Prometheus & Alert Manager is not providing any HA (instead, HAProxy should be used), but simply overwriting the previous value set.

Reply:
Right, as @epuertat suggested to push layouts you just need to update the grafana ep and execute the `dashboards update` command.
As @dsavineau pointed out, the endpoint configuration for HA using the VIP can be addressed on another thread.

Reply:
True, I'll update the PR